### ISSUE TITLE
Grad leakage in linesearch with scale_init_precond

### ISSUE THREAD

**Issue by younik** [google-deepmind/optax#1189]
Following https://github.com/google-deepmind/optax/pull/1133, linesearch uses the gradient norm to compute the identity scaling:

https://github.com/google-deepmind/optax/blob/8cb6a0e8797d05a0e5efbfc47bbd2851f8ae568b/optax/_src/transform.py#L1710

If we want to compute the gradient of the optimization process, this will make the gradient flow also from the identity scaling, which I believe is not intended.

## Suggestion

```diff
- capped_inv_norm = jnp.minimum(1.0, 1.0/otu.tree_l2_norm(updates)) 
+ capped_inv_norm = jnp.minimum(1.0, 1.0/otu.tree_l2_norm(jax.lax.stop_gradient(updates))) 
```

## MRE

This also makes the gradient less stable. In the following example, the gradient becomes NaN, while after the above change it is computed correctly.

```python
import jax
import jax.numpy as jnp
import optax


def run_optimization(
    init_params,
    fun,
    opt,
):

    value_and_grad_fn = optax.value_and_grad_from_state(fun)
    init_carry = (init_params, opt.init(params=init_params))

    def step(carry):

        params, state = carry

        value, grad = value_and_grad_fn(
            params,
            state=state,
        )

        updates, state = opt.update(
            updates=grad,
            state=state,
            params=params,
            value=value,
            grad=grad,
            value_fn=fun,
        )

        params = optax.apply_updates(params, updates)

        return params, state

    def continuing_criterion(carry):
        _, state = carry
        iter_num = optax.tree_utils.tree_get(state, "count")
        return iter_num < 10

    final_params, final_state = jax.lax.while_loop(
        continuing_criterion, step, init_carry
    )

    return final_params, final_state


A = jax.random.normal(jax.random.PRNGKey(0), (3,3))

def optimize(init_params):
    objective = lambda x: jnp.square(A @ x).sum()
    solution, _ = run_optimization(
        init_params=init_params,
        fun=objective,
        opt=optax.lbfgs(),
    )
    return init_params + solution


init_params = jnp.zeros(3)
opt_grad = jax.jacfwd(optimize)(init_params)
print(opt_grad)
```

**Comment by younik** [comment]
Solved with https://github.com/google-deepmind/optax/pull/1190
